actors/src/main/scala/zio/actors/Actor.scala

```diff
@@ -0,0 +1,115 @@
+package zio.actors
+
+import zio.actors.Actor.PendingMessage
+import zio._
+
+object Actor {
+
+  private[actors] type PendingMessage[E, F[_], A] = (F[A], Promise[E, A])
+
+  /**
+   *
+   * Description of actor behavior (can act as FSM)
+   *
+   * @tparam R environment type
+   * @tparam S state type that is updated every message digested
+   * @tparam E error type
+   * @tparam F message DSL
+   */
+  trait Stateful[R, S, +E <: Throwable, -F[+_]] extends AbstractStateful[R, S, E, F] {
```

Comment:
Does it have to be constrained by `Throwable`? I would either drop the constrain or fix it throughout the code.

Reply:
The reason we need `Throwable` here is because processing each message might fail with custom error of type `E` or some ActorSystem error: e.g the remote communication failure. So sending a message `!` to an actor is a `Task`. So to make `E` compatible I decided to put type bounds on this type to be included as possible failure result of`!(...): Task[...]`. (Random thought: I guess hypothetically we will avoid it with union types `E | Throwable` in dotty)

WDYT?
Is it correct argumentation?

Reply:
Ah so, it's there to accommodate a type coming from a lower level. Let's keep it as `RIO` then, and in the future we may want to design our own error hierarchy.

Reply:
Sure sounds good to me, so in that PR I'm also dropping `E` from the whole API (Stateful, Actor, ActorRef classes), ok?